test/e2e/aws_test.go

```diff
@@ -48,49 +59,75 @@ func (a *awsTestHelper) makeCredentialsSecret(namespace string) *corev1.Secret {
+			Source: operatorv1alpha1.ExternalDNSSource{
+				ExternalDNSSourceUnion: operatorv1alpha1.ExternalDNSSourceUnion{
+					Type: operatorv1alpha1.SourceTypeService,
+					Service: &operatorv1alpha1.ExternalDNSServiceSourceOptions{
+						ServiceType: []corev1.ServiceType{
+							corev1.ServiceTypeLoadBalancer,
+							corev1.ServiceTypeClusterIP,
+						},
+					},
+					AnnotationFilter: map[string]string{
+						"external-dns.mydomain.org/publish": "yes",
+					},
+				},
+				HostnameAnnotationPolicy: "Ignore",
+				FQDNTemplate:             []string{fmt.Sprintf("{{.Name}}.%s", hostedZoneDomain)},
+			},
+			Provider: operatorv1alpha1.ExternalDNSProvider{
+				Type: operatorv1alpha1.ProviderTypeAWS,
+				AWS: &operatorv1alpha1.ExternalDNSAWSProviderOptions{
+					Credentials: operatorv1alpha1.SecretReference{
+						Name: credsSecret.Name,
+					},
+				},
+			},
+		},
+	}
+}
 func (a *awsTestHelper) platform() string {
 	return string(configv1.AWSPlatformType)
 }
 
-func (a *awsTestHelper) ensureHostedZone(rootDomain string) (string, []string, error) {
+func (a *awsTestHelper) ensureHostedZone() (string, []string, error) {
 	zones, err := a.r53Client.ListHostedZones(&route53.ListHostedZonesInput{})
 	if err != nil {
 		return "", nil, fmt.Errorf("failed to list hosted zones: %w", err)
 	}
 
-	var (
-		nameServers  []string
-		hostedZoneID string
-	)
 	// if hosted zone exists then return its id and nameservers
 	for _, zone := range zones.HostedZones {
-		if aws.StringValue(zone.Name) == rootDomain {
+		if aws.StringValue(zone.Name) == hostedZoneDomain {
 			hostedZone, err := a.r53Client.GetHostedZone(&route53.GetHostedZoneInput{Id: zone.Id})
 			if err != nil {
 				return "", nil, fmt.Errorf("failed to get hosted zone %s: %v", aws.StringValue(zone.Id), err)
 			}
-			nameServers = aws.StringValueSlice(hostedZone.DelegationSet.NameServers)
-			hostedZoneID = aws.StringValue(zone.Id)
-			return hostedZoneID, nameServers, nil
+			return aws.StringValue(zone.Id), aws.StringValueSlice(hostedZone.DelegationSet.NameServers), nil
 		}
 	}
 
 	// create hosted zone and return its id and nameservers
 	zone, err := a.r53Client.CreateHostedZone(&route53.CreateHostedZoneInput{
-		Name:            aws.String(rootDomain),
+		Name:            aws.String(hostedZoneDomain),
 		CallerReference: aws.String(time.Now().Format(time.RFC3339)),
 	})
 	if err != nil {
 		return "", nil, fmt.Errorf("failed to get create hosted zone: %w", err)
 	}
-	nameServers = aws.StringValueSlice(zone.DelegationSet.NameServers)
-	hostedZoneID = aws.StringValue(zone.HostedZone.Id)
-	return hostedZoneID, nameServers, nil
+	return aws.StringValue(zone.HostedZone.Id), aws.StringValueSlice(zone.DelegationSet.NameServers), nil
 }
 
-func (a *awsTestHelper) deleteHostedZone(zoneID string) error {
+func (a *awsTestHelper) deleteHostedZone() error {
 	input := route53.ListResourceRecordSetsInput{
-		HostedZoneId: &zoneID,
+		HostedZoneId: &hostedZoneID,
```

Comment:
Same comment about the global variable usage.

Reply:
Done